.github/workflows/update-snapshot.yml

```diff
@@ -0,0 +1,91 @@
+
+on:
+  repository_dispatch:
+    types: update-data-snapshot
+
+env:
+  AWS_S3_BUCKET: content.covidactnow.org
+  BRANCH: ${{ github.event.client_payload.branch }}
+  SNAPSHOT_ID: ${{ github.event.client_payload.snapshot_id }}
+  # Used in the shared images URL and in the PR branch.
+  RUN_ID: ${{ github.event.client_payload.snapshot_id }}-${{github.run_number}}
+
+jobs:
+  update-data-snapshot:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - name: Verify snapshot provided
+        if: ${{ !env.SNAPSHOT_ID }}
+        run: 'echo "Missing client_payload parameter: snapshot_id" ; exit 1'
+      - name: Verify branch provided
+        if: ${{ !env.BRANCH }}
+        run: 'echo "Missing client_payload parameter: branch" ; exit 1'
+      - uses: actions/checkout@v2
+        with:
+          repository: covid-projections/covid-projections
+          ref: '${{ env.BRANCH }}'
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+
+      # Update snapshot URL and regenerate map colors.
+      - run: yarn install
+      - run: 'echo "{ \"data_url\": \"https://data.covidactnow.org/snapshot/${{env.SNAPSHOT_ID}}/\" }" | python -mjson.tool > src/assets/data/data_url.json'
+      - run: yarn update-calculated-interventions
+
+      # Generate Share Images and Upload to S3
+      - run: yarn serve -l 3000 &
+      - run: npx wait-on -t 120000 http://localhost:3000/
+      - run: yarn generate-share-images
+      - uses: jakejarvis/s3-sync-action@master
+        with:
+          args: '--acl public-read --follow-symlinks --delete'
+        env:
+          #TODO(michael): Make default secrets work.
+          #AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY }}
+          #AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.CONTENT_AWS_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CONTENT_AWS_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: scripts/generate_share_images/output
+          DEST_DIR: share/${{env.RUN_ID}}
+      - run: 'echo "{ \"share_image_url\": \"https://content.covidactnow.org/share/${{env.RUN_ID}}/\" }" | python -mjson.tool > src/assets/data/share_images_url.json'
+
+      # Generate Pull Request
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v2
+        with:
+          branch: 'bump-snapshot-${{env.RUN_ID}}'
+          commit-message: |
+            Update to data snapshot ${{env.SNAPSHOT_ID}}.
+
+            * Regenerate calculated interventions files.
+            * Point at updated share images.
+          title: Update to data snapshot ${{env.SNAPSHOT_ID}}
+          body: |
+            Update to data snapshot ${{env.SNAPSHOT_ID}}.
+
+            * Regenerate calculated interventions files.
+            * Point at updated share images.
+
+      - name: Slack notification
+        if: job.status != 'success'
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
```

Comment:
FYI- I just used the existing webhook secret that was configured for the repo.  If we want to send this somewhere other than #eng or we want the name to be different than "Publishing New API", I'll need some help setting it up.  But I think it's fine? 🤷

Reply:
yeah i was thinking we should change the name of that webhook anyway to be something more general?

Reply:
Yes, thanks!